### PR TITLE
Adds sample code to TextFormField

### PR DESCRIPTION
**Pull request by jslavitz**
Fixes nits from #11528 originally created by @branflake2267

### CODE REVIEW

File: packages/flutter/lib/src/material/text_form_field.dart [hunk 1]
```diff
@@ -38,6 +38,22 @@ import 'theme.dart';
...
+/// ## Sample code
+///
+/// ```dart
+/// new TextFormField(
+///   decoration: const InputDecoration(
+///     icon: const Icon(Icons.person),
+///     hintText: 'What do people call you?',
+///     labelText: 'Name *',
+///   ),
+///   onSaved: (String value) {},
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/text_form_field.dart]: The empty callback seems odd. Maybe describe in a comment what you would do in here?

Comment (jslavitz) [reply packages/flutter/lib/src/material/text_form_field.dart]: Ok, I added a bit of an explanation which I think should make the example a bit more clear!

File: packages/flutter/lib/src/material/text_form_field.dart [hunk 2]
```diff
@@ -38,6 +38,25 @@ import 'theme.dart';
 ///    integration.
 ///  * [InputDecorator], which shows the labels and other visual elements that
 ///    surround the actual text editing widget.
+///
+/// ## Sample code
+/// Creates a [TextFormField] with an [InputDecoration] and validator function.
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/text_form_field.dart]: nit: empty line after "Sample Code"

File: packages/flutter/lib/src/material/text_form_field.dart [hunk 3]
```diff
@@ -38,6 +38,25 @@ import 'theme.dart';
...
+/// In this example, onSaved is empty, but can optionally be used to call a
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/text_form_field.dart]: onSaved -> `onSaved`

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by tvolkert — APPROVED** [review]

**Pull request merged**